### DISCUSSION TITLE
REST Questions

### DISCUSSION THREAD

**Discussion by jmg011** [NetApp/harvest#1102] [Q&A] [answered]
We are starting the transition to REST api from ZAPI. I have following questions regarding the same.

1) Do all the Harvest plugins/collectors support REST api?
2) Apart from mapping the metrics in the templates _(including both default and custom templates I created)_ to REST API and building new templates with REST and updating harvest.yml with REST config, is there anything else that we have to do differently for Harvest set up with REST api?

**Comment by rahulguptajss** [comment, marked as answer]
@jmg011 This information is available [here](https://github.com/NetApp/harvest/blob/main/docs/architecture/rest-collector.md)
 
1. Yes all plugins/collectors are supported 9.12+. Most of the REST config templates will work before 9.12, but unless you have specific needs, we recommend sticking with the ZAPI collectors until you upgrade to 9.12.
2. Default templates are already mapped. Only change in collector (REST) is required in harvest.yaml.

**Note:** RestPerf collector will be communicated seperately as that is currently under development.
